pkg/roachprod/fluentbit/fluentbit.go

```diff
@@ -0,0 +1,104 @@
+package fluentbit
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"strings"
+	"text/template"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/errors"
+)
+
+//go:embed files/fluent-bit.yaml.tmpl
+var fluentBitTemplate string
+
+//go:embed files/fluent-bit.service
+var fluentBitSystemdUnit string
+
+type Config struct {
+	DatadogSite   string
+	DatadogAPIKey string
+}
+
+func Install(ctx context.Context, l *logger.Logger, c *install.SyncedCluster, config Config) error {
+	if err := install.InstallTool(ctx, l, c, c.Nodes, "fluent-bit", l.Stdout, l.Stderr); err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	var combinedErr error
+
+	// The Fluent Bit configuration needs the hostname of each node.
+	for _, node := range c.Nodes {
+		buf.Reset()
+
+		if err := c.Run(ctx, l, &buf, l.Stderr, install.WithNodes(install.Nodes{node}), "hostname", "hostname -f"); err != nil {
+			combinedErr = errors.CombineErrors(combinedErr, err)
+			continue
```

Comment:
Any non-nil error is over all considered a failure, right? Ang reason to not just early return right here / the cases below?